uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/crm/app.form.workpattern.js

```diff
@@ -417,28 +423,108 @@ define([
      * @private
      */
     Day.prototype._setFieldsMasks = function () {
-      var hourMask = Inputmask({
-        'mask': '99:99',
-        'oncomplete': this._calculateNumberOfHours.bind(this)
+      var self = this;
+
+      var timeFromMask = Inputmask({
+        mask: '99:99',
+        oncomplete: function () {
+          self._onTimeChange('From');
+        }
+      });
+
+      var timeToMask = Inputmask({
+        mask: '99:99',
+        oncomplete: function () {
+          self._onTimeChange('To');
+        }
       });
 
       var breakMask = Inputmask({
-        'alias': 'decimal',
-        'rightAlign': false,
-        'oncomplete': this._calculateNumberOfHours.bind(this)}
-      );
+        alias: 'decimal',
+        rightAlign: false,
+        oncomplete: function () {
+          self._calculateNumberOfHours();
+        }
+      });
 
       var numberOfHoursMask = Inputmask({
-        'alias': 'decimal',
-        'rightAlign': false
+        alias: 'decimal',
+        rightAlign: false,
+        oncomplete: this._onNumberOfHoursChange.bind(this)
       });
 
-      hourMask.mask(this._timeFromField);
-      hourMask.mask(this._timeToField);
+      timeFromMask.mask(this._timeFromField);
+      timeToMask.mask(this._timeToField);
       breakMask.mask(this._breakField);
       numberOfHoursMask.mask(this._numberOfHoursField);
     };
 
+    /**
+     * Reacts to the change of a time input of a given type.
+     * Adjusts the time value, the range between times, rounds the value
+     * and finally calculates the number of hours.
+     *
+     * @param {String} timeType From|To
+     */
+    Day.prototype._onTimeChange = function (timeType) {
+      this._adjustTimeValue(timeType);
+      this._adjustTimesRange(timeType);
+      this._roundTimeValue(timeType);
+      this._calculateNumberOfHours();
+    };
+
+    /**
+     * Gets the time input by a type
+     *
+     * @param  {String} timeType From|To
+     * @return {jQuery}
+     */
+    Day.prototype._getTimeInputOfType = function (timeType) {
+      return $(this['_time' + timeType + 'Field']);
+    };
+
+    /**
+     * Reacts to the change of number of hours.
+     * If the amount is greater than the range between times, sets to that range.
+     */
+    Day.prototype._onNumberOfHoursChange = function () {
+      var $numberOfHoursField = $(this._numberOfHoursField);
+      var rangeInHours = moment.duration(
+        moment(this._timeToField.value, timeFormat)
+          .diff(moment(this._timeFromField.value, timeFormat))).asHours();
+
+      if ($numberOfHoursField.val() > rangeInHours) {
+        this._calculateNumberOfHours(true);
+      }
+    };
+
+    /**
+     * Adjusts the range between times.
+     * If the range is less than 0, then adjusts the time input that was changed.
+     *
+     * @param {String} timeType From|To
+     */
+    Day.prototype._adjustTimesRange = function (timeType) {
```

Comment:
adjust how? if the explanation is too big, please add it to the description.

Reply:
The explanation on what exactly it does is in the function description: https://github.com/compucorp/civihr/pull/2856/files#diff-06deb86aea35fed830b02774a44a2e7bR501

I also mention it in the PR description as a comment to code